### ISSUE TITLE
Site Editor: Improve focus indication

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#47775]
### Description

Any interactive element that receives keyboard focus needs to have a clear focus indication.

In the Site Editor, there are a couple places where focus indication is missing:

1
In Browse mode, the active menu item doesn't have a focus style.

2
The preview on the right is an `iframe` element with `role="button"` and `aria-label="Editor canvas"`. It is focusable and it's in the tab sequence. When activated, the Editor switches to 'Edit mode'. Under all aspects, it's a keyboard operable interactive element. However, it doesn't have a focus style.

### Step-by-step reproduction instructions

- Go to the Site editor.
- Click 'Templates'.
- Click one of the templates in the navigation e.g. 'Page'.
- The 'Page' item is now the active (current) menu item in the navigation. It gets a blue background and white text.
- Use the Tab key to tab backwards and forwards through the navigation items.
- Observe that when tabbing to the active 'Page' menu item, its style doesn't change so that there's no focus indication.
- Press the Tab key to move focus to the 'Drag to resize' button. Visually, it's a vertical 'bar' that allows to resize the navigation sidebar.
- Press tab again.
- Keyboard focus is now on the template preview iframe with `role="button"`.
- Observe there's no focus style.

### Screenshots, screen recording, code snippet

In the screenshot below, the items highlighted with a red outline miss a focus style.

<img width="1792" alt="Screenshot 2023-02-06 at 10 50 40" src="https://user-images.githubusercontent.com/1682452/216945518-13322af9-ec94-4e04-a67c-3c235536d93c.png">

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
